lms/djangoapps/certificates/tests/test_api.py

```diff
@@ -405,3 +408,80 @@ def _assert_cert_status(self, *expected_statuses):
         """Check the example certificate status. """
         actual_status = certs_api.example_certificates_status(self.COURSE_KEY)
         self.assertEqual(list(expected_statuses), actual_status)
+
+
+def set_microsite(domain):
+    """
+    returns a decorator that can be used on a test_case to set a specific microsite for the current test case.
+    :param domain: Domain of the new microsite
+    """
+    def decorator(func):
+        """
+        Decorator to set current microsite according to domain
+        """
+        @wraps(func)
+        def inner(request, *args, **kwargs):
+            """
+            Execute the function after setting up the microsite.
+            """
+            microsite.set_by_domain(domain)
+            return func(request, *args, **kwargs)
+        return inner
+    return decorator
+
+
+@override_settings(FEATURES=FEATURES_WITH_CERTS_ENABLED)
+@attr('shard_1')
+class CertificatesBrandingTest(TestCase):
+    """Test certificates branding. """
+
+    COURSE_KEY = CourseLocator(org='test', course='test', run='test')
+
+    def setUp(self):
+        super(CertificatesBrandingTest, self).setUp()
+
+    @set_microsite(settings.MICROSITE_CONFIGURATION['test_microsite']['domain_prefix'])
+    def test_certificate_header_data(self):
+        """
+        Test that get_certificate_header from certificates api returns urls customized according to site branding.
```

Comment:
This test should confirm all possible context data generated by the method -- the docstring only mentions the urls.

Also, I think it would be good for this test to fail if 1) unexpected keys are found in the returned dictionary, or 2) expected keys are not found in the dictionary.  We can do this also for the footer and template tests.